src/clean.rs

```diff
@@ -96,4 +97,9 @@ pub fn clean_dom(dom: &NodeRef) {
 
         html::add_class(&a_el, "css");
     }
+
+    // Move <style> elements into the <head>.
+    for style_el in html::select(doc.body(), "style") {
```

Comment:
Can you clarify in the comment above why doing this rather than appending to the head directly? Is it to preserve ordering?

Reply:
Yes, I do this to preserve order, but that's only necessary for testing.